tests/unit/core/sagas/test_abuse.js

```diff
@@ -34,78 +39,160 @@ describe(__filename, () => {
+      const addon = { ...fakeAddon, guid: 'some-guid' };
 
-    const errorAction = errorHandler.createErrorAction(error);
-    await sagaTester.waitFor(errorAction.type);
-    expect(sagaTester.getCalledActions()[2]).toEqual(errorAction);
-  });
+      mockAddonManager
+        .expects('reportAbuse')
+        .withArgs(addon.guid)
+        .resolves(false);
 
-  it('throws an error if multiple reports are submitted for the same add-on', async () => {
-    _sendAddonAbuseReport({
-      addonSlug: 'some-addon',
-      message: 'This add-on is malwaré!',
+      _initiateAddonAbuseReportViaFirefox({ addon });
+
+      await sagaTester.waitFor(finishAddonAbuseReportViaFirefox().type);
+      mockAddonManager.verify();
     });
 
-    // Report the same add-on again; this will cause the reducer to throw
-    // an error and the saga should dispatch an error.
-    _sendAddonAbuseReport({
-      addonSlug: 'some-addon',
-      message: 'Duplicate!',
+    it('dispatches loadAddonAbuseReport after a successful report', async () => {
+      const addon = { ...fakeAddon, guid: 'some-guid' };
+
+      mockAddonManager.expects('reportAbuse').resolves(true);
+
+      _initiateAddonAbuseReportViaFirefox({ addon });
+
+      const expectedLoadAction = loadAddonAbuseReport({
+        addon: { guid: addon.guid, id: addon.id, slug: addon.slug },
+        message: null,
+        reporter: null,
+      });
+
+      const loadAction = await sagaTester.waitFor(expectedLoadAction.type);
+      expect(loadAction).toEqual(expectedLoadAction);
     });
 
-    await sagaTester.waitFor(SET_ERROR);
-    expect(sagaTester.getCalledActions()[1]).toEqual(
-      errorHandler.createClearingAction(),
-    );
+    it('dispatches finishAddonAbuseReportViaFirefox after a successful report', async () => {
+      mockAddonManager.expects('reportAbuse').resolves(true);
+
+      _initiateAddonAbuseReportViaFirefox();
+
+      const expectedAction = finishAddonAbuseReportViaFirefox();
+      const finishAction = await sagaTester.waitFor(expectedAction.type);
+      expect(finishAction).toEqual(expectedAction);
+    });
+
+    it('does not dispatch loadAddonAbuseReport after a cancelled report', async () => {
+      const addon = { ...fakeAddon, guid: 'some-guid' };
+
+      mockAddonManager.expects('reportAbuse').resolves(false);
+
+      _initiateAddonAbuseReportViaFirefox({ addon });
+
+      await sagaTester.waitFor(finishAddonAbuseReportViaFirefox().type);
+
+      const unexpectedAction = loadAddonAbuseReport({
+        addon: { guid: addon.guid, id: addon.id, slug: addon.slug },
+        message: 'Abuse report via Firefox',
+        reporter: null,
+      });
+      expect(sagaTester.numCalled(unexpectedAction.type)).toEqual(0);
+    });
+
+    it('dispatches finishAddonAbuseReportViaFirefox on error', async () => {
+      const error = new Error('An error from reportAbuse');
+      mockAddonManager.expects('reportAbuse').returns(Promise.reject(error));
```

Comment:
I am curious, would this work?

```suggestion
      mockAddonManager.expects('reportAbuse').rejects(error);
```

Reply:
Oh yeah, that should work. Thanks!